plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/internal/util/GenericsUtil.java

```diff
@@ -31,4 +36,79 @@ public static ParameterizedType getParametrizedType(final Class<?> clazz, final
          .getActualTypeArguments()[0];
    }
+
+   /**
+    * This method will search for an actual type argument that matches the expected base type, starting from the given
+    * clazz up the complete class hierarchy. If the given type object matches the actual type argument, the type object
+    * is returns as that type.
+    *
+    * @param <T>        expected base type
+    * @param clazz      search start
+    * @param baseType   base type that matches the actual type argument
+    * @param typeObject the object that may implement the given type
+    * @return the given type object as the matching type
+    */
+   public static <T> Optional<? extends T> asActualTypeArgument(final Class<?> clazz, final Class<T> baseType,
+      final Object typeObject) {
+      return findActualTypeArgument(clazz, baseType)
+         .filter(matchingType -> matchingType.isInstance(typeObject))
+         .map(matchingType -> matchingType.cast(typeObject));
+   }
+
+   /**
+    * This method will search for an actual type argument that matches the expected base type, starting from the given
+    * clazz up the complete class hierarchy.
+    *
+    * @param <T>      expected base type
+    * @param clazz    search start
+    * @param baseType base type that matches the actual type argument
+    * @return the type argument that is closest to the given <code>clazz</code> and matches the given base base
+    */
+   public static <T> Class<? extends T> getActualTypeArgument(final Class<?> clazz, final Class<T> baseType) {
+      return findActualTypeArgument(clazz, baseType)
+         .orElseThrow(() -> new GLSPServerException("No matching type argument for " + baseType + " in " + clazz));
+   }
+
+   /**
+    * This method will search for an actual type argument that matches the expected base type, starting from the given
+    * clazz up the complete class hierarchy.
+    *
+    * @param <T>      expected base type
+    * @param clazz    search start
+    * @param baseType base type that matches the actual type argument
+    * @return the type argument that is closest to the given <code>clazz</code> and matches the given base base
+    */
+   public static <T> Optional<Class<? extends T>> findActualTypeArgument(final Class<?> clazz,
+      final Class<T> baseType) {
+      return findActualTypeArgument(clazz, baseType, null);
+   }
+
+   /**
+    * This method will search for an actual type argument that matches the expected base type, starting from the given
+    * clazz until the search stop.
+    *
+    * @param <T>        expected base type
+    * @param clazz      search start
+    * @param baseType   base type that matches the actual type argument
+    * @param searchStop search stop or <code>null</code> if we should search up to Object
+    * @return the type argument that is closest to the given <code>clazz</code> and matches the given base base
+    */
+   @SuppressWarnings("unchecked")
```

Comment:
There seems to be a checkstyle warning. Could you please have a look at it?

Reply:
Also done, somehow I broke my local checkstyle configuration. Do you remember which version of checkstyle must be installed?

Reply:
We use `emfcloud-checkstyle-8.44.xml` config. So I guess it should work with checkstyle 8.44 :wink: